src/Lean/Data/Name.lean

```diff
@@ -184,6 +184,11 @@ def anyS (n : Name) (f : String → Bool) : Bool :=
   | .num p _ => p.anyS f
   | _ => false
 
+/-- Return true if the name is in a namespace associated to metaprogramming.-/
+def isMetaprogramming (n : Name) : Bool :=
+  let components := n.components
+  components.head? == `Lean || (components.any fun n => n == `Tactic || n == `Linter)
```

Comment:
The environment linters in Batteries use `Batteries.Tactic.Lint` as part of their namespace: they get excluded by `Tactic`, but maybe you could
* also filter `Lint` here, or
* change their namespace to `Linter` instead of `Lint`.